### ISSUE TITLE
Data Ingest: calculate md5 sums

### ISSUE THREAD

**Issue by coleshaw** [mountetna/monoetna#935]
For data ingested from the CAT to C4 / Metis, it would be nice if:

* Generate an md5.txt list for all files ingested to the CAT.
* Check md5's against Metis to see what has already been uploaded, and remove from the CAT if so (in lieu of just size-updated-at-created-at)

**Comment by dtm2451** [comment]
Borne of thinking around a current use-case for me.

A file listing md5s of all files in the folder would be a minor value-add for projects' DSs -- a time-saver in cases where fastqs are re-processed by the CAT Core and then the DS must determine what files are actually updated.  Sometimes, only a few samples of a whole bunch (say 2/4) will actually change at all.  That info then tells me what samples to re-run through down stream processing steps.

Bullet 2 - Removing such files from c4 might not always be the right move.  If the DS had not yet touched their files (and specifically, had not moved their previous files from staging to elsewhere) then doing so might still remove the original/only copy.

**Comment by corps** [comment]
Can we also include, recursively, the md5 sum of each the contents of each child directory as well?
this would allow us to even track when inner directories have changed easily as well with a single computation.

**Comment by graft** [comment]
I'm not sure I understand bullet 1 - don't we already generate MD5s when files are put on Metis?

**Comment by coleshaw** [comment]
The list of MD5's is not available on C4 or in any other "readily apparent" form ... so someone would have to either re-calculate or run some sort of script against Metis to see the md5 hashes...

**Comment by corps** [comment]
That's a use case for a metis -> metis etl process, right?

**Comment by coleshaw** [comment]
Yeah, so our current use case is that the CAT re-processed and re-uploaded a set of files, of which only a subset actually changed. But the current ETL process has no way of knowing that (it just uses size-updated-at-created-at as a pseudo-hash to detect file changes), since it sees a new updated-at timestamp. We were thinking:

1) It would be nice if there were some way to check md5's against some known set (i.e. Metis), to prevent duplicate files and additional human filtering work. This necessitates downloading the file somewhere first, since we stream it from the CAT (cannot use the container running the process). If we're calculating it, we could also store the md5's on C4 at the same time, for convenience.
2) DS might use the MD5's downstream for other tasks.

**Comment by corps** [comment]
Let me know if and how I can help here.
